internal/config/yaml_parser.go

```diff
@@ -42,9 +42,16 @@ func IsValidLogSeverity(severity LogSeverity) bool {
 		WARNING,
 		ERROR,
 		OFF:
-		return true
+		return nil
 	}
-	return false
+	return fmt.Errorf("error parsing config file: log severity should be one of [trace, debug, info, warning, error, off]")
+}
+
+func validateWriteConfig(writeConfig WriteConfig) error {
```

Comment:
Nit: The name is a bit redundant as it's taking in a WriteConfig, so it's obvious that it'll validate a WriteConfig. This will lead to code like 

```go
WriteConfig writeConfig = ...
validateWriteConfig(writeConfig)
```

 We could switch this to `func (writeConfig WriteConfig)Validate() error {` , which will reduce the above to

```go
writeConfig.validate()
```

Same goes for validateLogSeverity .

Reply:
Raised separate PR for all the suggested changes https://github.com/GoogleCloudPlatform/gcsfuse/pull/1412. PTAL

Thanks!